content/languages/javascript/authoring.md

```diff
@@ -0,0 +1,281 @@
+
+```javascript
+chai.config.truncateThreshold = 0;
+```
+
+:::
+
+### Generating random numbers
+
+JavaScript provides a simple [`Math.random`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random) function that returns a random floating-point number between `0` and `1`. You can use this to generate random numbers by multiplying it by another number, like `5`, and then flooring it, which returns a random number between `0` and **4**. Here's a simple utility to generate a random number between a lower and upper limit, inclusive:
+
+```javascript
+function random(lower, upper) {
+  return Math.floor(Math.random() * (upper - lower + 1) + lower);
+}
+
+random(1, 10); // inclusive random number between 1 and 10
+```
+
+However, to make this simpler, the [lodash](https://lodash.com) library includes some helpful functions to ease this:
+
+- [`_.random(lower, upper[, floating])`](https://lodash.com/docs#random) - Generates a random number between `lower` and `upper`, inclusively.
+- [`_.shuffle(array)`](https://lodash.com/docs#shuffle) - Returns shuffled `array`.
+- [`_.sample(array)`](https://lodash.com/docs#sample) - Returns a random element from `array`.
+- [`_.sampleSize(array, size)`](https://lodash.com/docs#sampleSize) - Returns a random number of elements up to `size` from `array`.
+
+### Additional packages
+
+The Codewars runner provides a set of preinstalled packages, which are available not only for users solving a kata, but can be also used by authors to build tests and generators of test cases.
+
+### Reference solution
+
+_TODO_
+
+### Calling assertions
+
+_TODO_
+
+## Example test suite
+
+Below you can find an example test suite that covers most of the common scenarios mentioned in this article. Note that it does not present all possible techniques, so actual test suites can use a different structure, as long as they keep to established conventions and do not violate authoring guidelines.
+
+```javascript
+const chai = require('chai');
+const assert = chai.assert;
+const _ = require('lodash');
+chai.config.truncateThreshold = 0;
+
+describe('Fixed tests', () => {
+  it('Regular cases', () => {
+    assert.strictEqual(6, userSolution([1, 2, 3]));
+    assert.strictEqual(5, userSolution([2, 3]));
+  });
+
+  it('Edge cases', () => {
+    assert.strictEqual(0, userSolution([]), "Invalid answer for empty array");
+    assert.strictEqual(2, userSolution([2]), "Invalid answer for one element array");
+  });
+
+  it('Input should not be modified', () => {
+    const arr = _.shuffle(_.range(1, 100));
+    const arrCopy = [...arr];
+    // call user solution and ignore the result
+    userSolution(arrCopy);
+    // arrCopy should not be modified
+    assert.deepEqual(arrCopy, arr, 'Input array was modified');
+  });
+});
+
+describe('Random tests', () => {
+  function referenceSolution(arr) {
```

Comment:
I believe the related paragraph is yet to come, but it would be worth to somehow stress in the example that it's important to put reference solutions in a limited scope.

```suggestion

  //reference solution enclosed in a local scope of `describe`
  function referenceSolution(arr) {
```